akka-http2-support/src/main/scala/akka/http/impl/engine/http2/AlpnSwitch.scala

```diff
@@ -75,11 +82,15 @@ object AlpnSwitch {
               .run()(interpreter.subFusingMaterializer)
           }
 
-          // helpers to connect inlets and outlets
+          // helpers to connect inlets and outlets also binding completion signals of given ports
           def connect[T](in: Inlet[T], out: SubSourceOutlet[T], initialElement: Option[T]): Unit = {
             val propagatePull =
               new OutHandler {
-                def onPull(): Unit = pull(in)
+                override def onPull(): Unit = pull(in)
+                override def onDownstreamFinish(): Unit = {
```

Comment:
I don't think these are necessary. No need to complete an `OutHandler`, that was just cancelled. Or am I missing something?